### PR TITLE
Add github actions & remove travis-ci

### PR DESCRIPTION
**Pull request by asherf**
Travis-CI is not in a good place.
many open source projects have moved away from it.
this PR removes the travis ci config (which stopped working a while ago) and adds GH actions.
The only issue I am unable to fix is the `testBigGetMulti` is failing in CI.
reducing the count to 10**3 fixes the issues so something tells me we are hitting some kind of limit.
However, I am not that familiar with memcached and pylibmc to be able to debug this, so any guidance here will be appreciated.

You can see the issue in my fork: https://github.com/asherf/pylibmc/actions/runs/1430630588
specifically:  https://github.com/asherf/pylibmc/runs/4129034995?check_suite_focus=true#step:10:71
Screen Shot 2021-11-06 at 8 52 27 PM

Other than this issue, all other tests pass:

Screen Shot 2021-11-06 at 8 52 58 PM

Fixes: https://github.com/lericson/pylibmc/issues/272

### CODE REVIEW

**Comment by asherf** [comment]
@lericson I am happy to incorporate any feedback you might have.

**Comment by timgraham** [comment]
I haven't been able to figure out why the test might be failing. I have no issue with memcached 1.5.22 locally. It seems like the test is failing because values are evicted from the cache on CI but both my machine and CI show `STAT maxbytes 67108864`.

**Comment by asherf** [comment]
> I haven't been able to figure out why the test might be failing. I have no issue with memcached 1.5.22 locally. It seems like the test is failing because values are evicted from the cache on CI but both my machine and CI show `STAT maxbytes 67108864`.

that was my guess as well, then I looked at the values in the Travis runs (from a few months back when they still worked) and those seem the same. I.e. maxmbytes is 64mb
Also note that under the test we get non of the values we requested (empty dict) , if some where evicted, wouldn't we get at least some values?

**Comment by timgraham** [comment]
I manually installed the latest memcached (1.6.12) on my machine and get this error:
```
======================================================================
ERROR: testBigGetMulti (tests.test_cmemcached.TestCmemcached)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tim/code/pylibmc/tests/test_cmemcached.py", line 58, in testBigGetMulti
    result = self.mc.get_multi(keys)
pylibmc.SomeErrors: error 19 from memcached_mget: (0x564136d71070) SYSTEM ERROR(Broken pipe),  host: 127.0.0.1:11211 -> libmemcached/io.cc:392
```
As you said, reducing the count to 10**3 resolves the issue. Investigation to continue...

**Comment by timgraham** [comment]
I put `get_multi()` calls inside the test's loop:
```diff
diff --git a/tests/test_cmemcached.py b/tests/test_cmemcached.py
index 97e7c53..b921a86 100644
--- a/tests/test_cmemcached.py
+++ b/tests/test_cmemcached.py
@@ -50,13 +50,19 @@ class TestCmemcached(PylibmcTestCase):
         # only contain bytes that can be implicitly decoded as ASCII.
         keys = ['key%d' % i for i in range(count)]
         pairs = zip(keys, ['value%d' % i for i in range(count)])
-
+        subkeys = []
         d = {}
         for key, value in pairs:
             d[key] = value
+            subkeys.append(key)
             self.mc.set(key, value)
-        result = self.mc.get_multi(keys)
-        eq_(result, d)
+            result = self.mc.get_multi(subkeys)
+            try:
+                eq_(result, d)
+            except AssertionError:
+                print(key)
+                print(value)
+                raise
 
     def testFunnyDelete(self):
         s = ""
```


 and found that it works correctly until the 2187th iteration (keys start at key0, so key2186) at which point `get_multi()` returns nothing. It would be interesting to see if the same iteration fails on CI.

Anyway, I think we should just update the test to use a smaller count as I think this is more likely an issue in memcached than in pylibmc.

**Comment by asherf** [comment]
updated based on @timgraham suggestion (thanks for looking into it)
and tests seem to pass: https://github.com/asherf/pylibmc/actions/runs/1441134908

**Comment by asherf** [comment]
@lericson ping
any feedback here? Happy to incorporate any feedback you have.

**Comment by timgraham** [comment]
Looks good to me. I think you could squash those commits.

You could try adding support for Python 3.10 (https://github.com/lericson/pylibmc/pull/271) in a separate commit. nose isn't compatible but maybe nose-py3 would be a drop in replacement.

Do you know what needs to be done so the workflow runs here?

**Comment by asherf** [comment]
@timgraham squashed.
I am happy to add support to 3.10, but I think it would make sense to have that in a follow up PR once this is merged (I'll do it).
I can also try to move things away from nose and to use pytest, that would probably help too...
But I think landing this PR and bringing back CI to this repo should be done first.
I think once this PR lands CI github actions will automatically enable.
It doesn't run now since this change is not on master.

**Comment by asherf** [comment]
@lericson ping

**Comment by timgraham** [comment]
I'd move "& reduced num of keys tested in testBigGetMulti" to a separate line in the commit message with more information. Something like:
```
The number of keys in testBigGetMulti is reduced because the larger number
doesn't work with memcached 1.6.12. For an unknown reason, get_multi()
with more than 2186 keys here returns nothing.
```

**Review by timgraham — APPROVED** [review]
I think "Reduced num of keys tested in testBigGetMulti" can be removed from the commit message.

Maybe GitHub actions needs to be enabled on this repo in order for the jobs to run. When I've added new jobs in a PR, they've run there without needing to be merged first.

**Comment by asherf** [comment]
@lericson ping

**Comment by asherf** [comment]
@lericson ping

**Comment by lericson** [comment]
Why was this closed and removed?

**Comment by lericson** [comment]
I will merge this even so.
